src/lib.rs

```diff
@@ -143,7 +124,7 @@ impl<A> ObfString<A> {
 		ObfString { key, data }
```

Comment:
might want to make new() `doc(hidden)` as well, to give us some room to further tweak its bounds without creating another stability hazard.

Reply:
You're right that eg. `ObfString::new(0, vec![1, 2, 3])` would break horribly when the decrypt method gets its hand on it as it would construct a `ObfBuffer<Vec<u8>>` from uninitialized data which is a big no-no.

I'm thinking of ways to avoid the uninit stuff entirely by eg cloning the obfuscated string and deobfuscating it inplace. It's slightly less efficient but would make the above completely safe.

Reply:
#25 does this. I do still think that new() should be removed from the publicly documented interface though. It's a dangerous method that limits what we can do to the internal representation of the types. We should make the new function `doc(hidden)` and tell people to go through the `obfconst` macro instead.

Reply:
If it's okay I've chosen to go with the clone() method which should keep it safe to expose this new method. stable0.1 is just meant to fix the bootstrap hack. Soon (tm) Rust will probably stabilize `min_const_generics` at which point the 0.2 version will not need to bootstrap hack anymore either and 0.1 is officially dead to me.

Reply:
Yeah, I just saw the tracking issue for `min_const_generics` showing some pretty good progress. It's closer than I expected.

The clone method is definitely fine. I took a casual look at the resulting assembly, it looks like the optimizer is able to see through it and avoids the memcpy anyways. So it shouldn't impact performance at all.

Thanks a lot for taking the time to work on a 0.1.2 release without the bootstrap hack, it's greatly appreciated!